cpp/src/parquet/file_reader.cc

```diff
@@ -638,11 +635,12 @@ class SerializedFile : public ParquetFileReader::Contents {
       const std::shared_ptr<Buffer>& footer_buffer, const uint32_t metadata_len,
       std::shared_ptr<InternalFileDecryptor> file_decryptor);
 
-  std::string HandleAadPrefix(FileDecryptionProperties* file_decryption_properties,
-                              EncryptionAlgorithm& algo);
+  std::string HandleAadPrefix(
+      const std::shared_ptr<FileDecryptionProperties>& file_decryption_properties,
+      EncryptionAlgorithm& algo);
```

Comment:
Ok, can we take the opportunity to change the mutable ref argument to const-ref?
```suggestion
  std::string HandleAadPrefix(
      const std::shared_ptr<FileDecryptionProperties>& file_decryption_properties,
      const EncryptionAlgorithm& algo);
```